docs: add topic guide for resource pools [DET-4348] (#1668)
determined-ai/determined: docs/topic-guides/system-concepts/resource-pool.txt

.. _resource-pool:

###############
 Resource Pool
###############

Large-scale deep learning requires using different resources, including
computation resources, memory, network, and storage. Typically, we start
off prototyping our models with immediately available instance types
with few accelerators. Once our models are mature enough, we want to

> scale up the training speed with stronger instance types with full

Reviewer: ```suggestion
scale up the training speed with larger instance types that have more powerful
```

accelerators and high-performing processors and more instances. In the

> meantime, in order to avoid the network becomes a bottleneck we want to

Reviewer: ```suggestion
meantime, in order to prevent the network from becoming a bottleneck we want to
```

> let the computation happen as close to data as possible. We also want to

Reviewer: I don't fully understand this line. Latency impacts throughput, but I didn't think it was that huge. I would probably remove the line about data locality

Author: removed

> use instance types with only CPUs to run TensorBoard or any bash

Reviewer: ```suggestion
use instance types without expensive accelerators to run TensorBoard or any bash
```

> commands to avoid wasting the usage of accelerators.

Reviewer: ```suggestion
commands to avoid wasting accelerators.
```

Determined provides the abstraction of physical resources for users to
choose on their tasks. We call this abstraction *resource pools*. We can
get the most value of our money by choosing different resource pools for
different goals.

Each resource pool is intended to map resources that are physically
close to each other. Each resource pool can be set to use one cloud
provider and one instance type and limited to have one region on AWS or

> zone on GCP. Each resource pool has an independent scheduling and

Reviewer: ```suggestion
zone on GCP. Each resource pool handles scheduling and scaling independently.
```

> scaling logic unit to make sure that Determined responds to any change

Reviewer: ```suggestion
```

> on the tasks and resources timely.

Reviewer: ```suggestion
```
